### PR TITLE
Replace "C Standard" with "ISO/IEC 9899:2018" and define "C standard library"

### PR DESCRIPTION
**Pull request by tkoeppe**
Fixes ISO/CS 035 (C++23 DIS).

Partially addresses https://github.com/cplusplus/nbballot/issues/544.
Fixes https://github.com/cplusplus/nbballot/issues/577.

### CODE REVIEW

File: source/basic.tex [hunk 1]
```diff
@@ -4728,9 +4728,8 @@
 \defn{value}, which is one discrete element of an
 \impldef{values of a trivially copyable type} set of values.
 \begin{footnote}
-The
-intent is that the memory model of \Cpp{} is compatible
-with that of ISO/IEC 9899 Programming Language C.
+The intent is that the memory model of \Cpp{} is compatible
+with that of \IsoCUndated{}.
```
Comment (zygoloid) [inline source/basic.tex:4732]: I think this is talking about the programming language, not about the ISO publication, so maybe
```suggestion
with that of the C programming language.
```

File: source/lib-intro.tex [hunk 2]
```diff
@@ -3715,8 +3714,7 @@
 Functions from the C standard library shall not throw exceptions%
 \indextext{specifications!C standard library exception}%
 \begin{footnote}
-That is, the C
-library functions can all be treated as if they
+That is, the C library functions can all be treated as if they
```
Comment (zygoloid) [inline source/lib-intro.tex:3717]:
```suggestion
That is, the C standard library functions can all be treated as if they
```
Use exactly the term we define in Clause 3, not a variant of it, per ISO rules.

Comment (tkoeppe) [reply source/lib-intro.tex:3717]: Good catch!

File: source/basic.tex (hunk 1 above)
Comment (tkoeppe) [reply source/basic.tex:4732]: Sounds good, thanks!

File: source/intro.tex [hunk 3]
```diff
@@ -12,9 +12,9 @@
 
 \pnum
 \Cpp{} is a general purpose programming language based on the C
-programming language as described in ISO/IEC 9899:2018
-\doccite{Programming languages --- C} (hereinafter referred to as the
-\defnx{C standard}{C!standard}). \Cpp{} provides many facilities
+programming language as described in \IsoC{}
+\doccite{Programming languages --- C}.
```
Comment (jensmaurer) [inline source/intro.tex:16]: We're not allowed to add a title next to the number. This needs to go.

File: source/lib-intro.tex [hunk 4]
```diff
@@ -150,9 +150,9 @@
 if it raises a floating-point exception other than \tcode{FE_INEXACT}.
 The semantics of a call to a C standard library function
 evaluated as a core constant expression
-are those specified in Annex F of the C standard
+are those specified in \IsoC{}, Annex F
 \begin{footnote}
-See also ISO/IEC 9899:2018 section 7.6.
+See also \IsoC{}, 7.6.
```
Comment (jensmaurer) [inline source/lib-intro.tex:155]: Should we add "subclause" before "7.6" ?

File: source/lib-intro.tex [hunk 5]
```diff
@@ -1599,8 +1599,7 @@
 \begin{footnote}
 The only reliable way to declare an object or
 function signature from the C standard library is by including the header that
-declares it, notwithstanding the latitude granted in 7.1.4 of the C
-Standard.
+declares it, notwithstanding the latitude granted in \IsoC{}, 7.1.4.
```
Comment (jensmaurer) [inline source/lib-intro.tex:1602]: "subclause"?

There are citation examples in the House Style, I think.

File: source/support.tex [hunk 6]
```diff
@@ -6164,6 +6164,6 @@
 The header \libheader{stdlib.h}
 assuredly provides the same declarations and definitions within
 the global namespace,
-much as in the C Standard. It may also provide these names within
+much as in \IsoC{}. It may also provide these names within
```
Comment (jensmaurer) [inline source/support.tex:6167]: Make that undated? We don't refer to a specific clause/annex.

File: source/lib-intro.tex (hunk 4 above)
Comment (tkoeppe) [reply source/lib-intro.tex:155]: Not according to the House Style examples of citing specific elements, I think.

File: source/support.tex (hunk 6 above)
Comment (tkoeppe) [reply source/support.tex:6167]: Hm, we also don't have an undated entry in the normative references, which is perhaps a problem: If we have both dated and undated ones, we might end up depending  on multiple versions of C. Maybe we can avoid the undated references entirely?

File: source/lib-intro.tex (hunk 4 above)
Comment (tkoeppe) [reply source/lib-intro.tex:155]: """
Use the order of “document number:date, element” when referring to a specific element:

“…in accordance with ISO 12345:2018, 4.3.9.”
“…in accordance with ISO/IEC TR 6789:2020, Table 3.”
“…in accordance with ISO 12345:2018/Amd 1:2019, Figure C.1.”
"""

Only things like "Table", "Figure", or "Annex" (and presumably "Clause") get a name.

File: source/support.tex (hunk 6 above)
Comment (jensmaurer) [reply source/support.tex:6167]: This is a note / example.  Everywhere else we use undated refs, we are also non-normative or explanatory, I think. So it seems we're good.

File: source/intro.tex (hunk 3 above)
Comment (tkoeppe) [reply source/intro.tex]: Done.

File: source/support.tex (hunk 6 above)
Comment (tkoeppe) [reply source/support.tex:6167]: OK, let's try. We do actually use an undated C reference normatively in one place (with the floating point types) :-S

**Review by jensmaurer — APPROVED** [review]

**Pull request merged**